MsCorePkg/HelloWorldRustDxe/src/main.rs

```diff
@@ -8,39 +8,56 @@
 //!
 //! SPDX-License-Identifier: BSD-2-Clause-Patent
 //!
-#![no_std]
-#![no_main]
+#![cfg_attr(target_os = "uefi", no_std)]

```

Comment:
Within the context of this PR, this is obvious, but since this is a sample module, I think an explanatory comment describing that this is done to support `no_std` for UEFI builds while also allowing `std` to be used for unit testing would be helpful for people looking at it.

Reply:
Should also help them track the pattern throughout the rest of the file.

Reply:
Technically, I think you could allow no_std for purposes of testing, though I haven't tried it; the important ones for enabling test builds are no_main and the panic handler being gated. I'll experiment a bit with whether no_std actually needs to be gated and update the PR.

Reply:
Updated - removed target gate on no_main, and added a descriptive comment.